### ISSUE TITLE
The last version cjxl.exe can't turn jpg files to lossy jxl files.

### ISSUE THREAD

**Issue by brucekiller** [libjxl/libjxl#1706]
**Describe the bug**
if input is jpg files, cjxl only output lossless jxl files,with -q 、-d options.

**Screenshots**
![image](https://raw.iqiq.io/brucekiller/images/a137798d6cdfb5ac754f0d1bb132aaa1fb34c20e/1618.jpg)
[AVX2,SSE4,SSSE3,Unknown]("Unknown" is "Emu128" in v0.7.0)

**Environment**
 - OS:  Windows10 x64 21H2
 - CPU type:  x86_64
 - cjxl/djxl version string:  cjxl v0.8.0 aa5e8f0

**Comment by mo271** [comment]
Could you try with `--lossless_jpeg=0`, e.g. 
`R:\>cjxl.exe --lossless_jpeg=0 -q 80 R:\3.jpg 3.jxl`

**Comment by brucekiller** [comment]
@mo271 Thanks，the --lossless_jpeg=0 can work. But I think the way of working in version 0.7.0 is better, the default is the same lossless. when I need lossy output, just need to use the option "-q or -d".

**Comment by mo271** [comment]
@brucekiller thanks for bringing this up. 
If I recall correctly, it was a conscious decision to make it more cumbersome to apply lossy compression to a jpg (or git for that matter), because they are already lossy and we don't want users to accidentally introduce yet more lossy compression to them. 
Perhaps it would be helpful to add a warning whenever the "-d" or "-q" are ignored?
What do you think, @jonsneyers?

**Comment by jonsneyers** [comment]
I don't think we should ever ignore an explicit -d/-q, but I do think it makes sense to have a different default -d/-q depending on how the input is given (default to lossless recompression if already lossy, d1 lossy if presumably pristine). When someone passes an explicit -d 3 and gives input as a (very high quality, e.g. straight from a camera) jpeg, it does make sense to apply lossy compression to the decoded pixels, and not do lossless jpeg recompression.

Orthogonally to that: we could try to do something clever when specifying an explicit -d/-q: e.g. when someone does a -d 0.5 to a jpeg input that is using quantization tables that suggest it is very lossy, it is likely that lossless recompression will be more effective than doing a very high quality lossy recompression of decoded pixels. The same could be true for e.g. png8 input that uses a palette of only 4 colors: it is quite likely that lossless compression will be better than lossy. Basically we could heuristically try to identify some cases where lossy might produce a file that is larger than lossless (whether lossless jpeg recompression or lossless pixels), and try both in those cases, returning the smallest one.
(another option could be to add an option --also_try_lossless that always does this, or to always try it if the distance target is below some threshold).
One question is if the logic for such behavior should be in libjxl or in cjxl; libjxl does not know where the pixels are coming from so it has less information to make the decision than cjxl, but adding the logic in libjxl means other applications can also benefit from it (e.g. when someone draws something in Krita or Gimp, it depends on the image contents whether lossy or lossless will be most effective). In any case it is quite silly to produce e.g. a 500 KB lossy file if you can also produce a 300 KB lossless file. Of course the compute cost to always try both is a problem in many cases, but we could tie it to the effort setting, or have an option to control the extra cost (e.g. 0 = never try it, 100 = always try it, in between = conditionally try it if some cheap heuristics say that it might be worth trying).
